src/index-widget/index-widget.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Component,
+  Input,
+  Inject,
+  forwardRef,
+  Optional,
+  SkipSelf,
+} from '@angular/core';
+import { Widget, BaseWidget } from '../base-widget';
+import {
+  InstantSearchInstance,
+  NgAisInstantSearch,
+} from '../instantsearch/instantsearch';
+import indexWidget, { Index } from 'instantsearch.js/es/widgets/index/index';
+
+const connectIndex = () => indexWidget;
+
+@Component({
+  selector: 'ais-index',
+  template: `<ng-content></ng-content>`,
+})
+export class NgAisIndex extends BaseWidget {
+  @Input() public indexName: string;
+  @Input() public indexId?: string;
+
+  public widget?: Index;
+
+  constructor(
+    @SkipSelf()
+    @Inject(forwardRef(() => NgAisIndex))
+    @Optional()
+    public indexParent: NgAisIndex,
+    @Inject(forwardRef(() => NgAisInstantSearch))
+    public instantSearchParent: NgAisInstantSearch
+  ) {
+    super('Index');
+  }
+
+  public addWidgets(widgets: Widget[]) {
+    this.widget.addWidgets(widgets);
+  }
+
+  public removeWidgets(widgets: Widget[]) {
+    this.widget.removeWidgets(widgets);
+  }
+
+  ngOnInit() {
+    this.createWidget(connectIndex, {
+      indexName: this.indexName,
```

Comment:
Is there a way to forward all the props? This way we make sure to pass everything that the InstantSearch `index` widget needs once we support more props.

Reply:
no, in Angular there's not really a concept of general props, but only class members with the @Input decorator as far as I can tell